### DISCUSSION TITLE
How do you create combat data files?

### DISCUSSION THREAD

**Discussion by Synexcu** [XVs32/kcauto_custom#71] [Q&A] [answered]
Hello, I just want to say thanks for your help on previous discussion. I already figured it out on why it was stuck.

Now, I would like to as a new question, How do you create combat data files? Like, How do I get the coords for the nodes, How do you get A, B, C, A3, A4, B2, C3, etc. on the edges (from your previous E-7 combat data) and what are the differences. Is there like an application that you use to map all of those coords?

Thank you

**Comment by XVs32** [comment, marked as answer]
Hi [Synexcu](https://github.com/Synexcu), 

Yes, I did made a simple tool to help mapping all those coords.
A similar question was ask in [issue#20](https://github.com/XVs32/kcauto_custom/issues/20).

Anyway, I wrote a [wiki page](https://github.com/XVs32/kcauto_custom/wiki/Ch99:-Appendix) for ppl who wants to make their own map support.  
You can check it out.

**Comment by Synexcu** [comment]
Hello,

I tried running `./tool/get_mouse_location.py` but the app can't run. I double clicked and it just showed a window for not even a second, and disappeared. Any guide on how to open it?

**Comment by XVs32** [comment]
Hi [Synexcu](https://github.com/Synexcu),

Sadly the `./tool/get_mouse_location.py` is not an exe/app. It is a Python source code file.
You do have to install Python(and libraries) on your computer to run it.
There should be tons of [tutorials](https://www.youtube.com/watch?v=Kn1HF3oD19c&ab_channel=AmitThinks) around.

Your end result should looks something like this:
![result](https://github.com/XVs32/kcauto_custom/assets/16824564/f00dcf74-4f7b-40ac-ab23-727588103dcf)

Good luck

**Comment by Synexcu** [comment]
Oooh, so that's how you run it. Got it, thanks!

After installing the lib, it works fine for me.

![image](https://github.com/XVs32/kcauto_custom/assets/82300898/c715b7ce-e41c-492e-85e8-d6482e2b6f08)

For the `You click top left of the map, then click from A to Z, it will shows you the X/Y coordinate then you can copy(and edit) the result."` I still don't understand that part. Could you elaborate more and perhaps show which part of `map` does it refer to? Thanks again

**Comment by XVs32** [comment]
Sure, here it is:

![top_left](https://github.com/XVs32/kcauto_custom/assets/16824564/4f2b89b1-4d2e-4d7c-88b8-0ea7dcac6dc0)

Click the top left (red circle) first, then the nodes from A~Z (yellow circle)

**Comment by Synexcu** [comment]
Amazing! Thanks for the answer!. For the coords I assume we have to click on the middle most part of the node, yes?

**Comment by XVs32** [comment]
Where you click is where kcauto will click when node select (etc.) is needed.
So yes, middle part of the node, it doesn't have to be 100% precise.

**Comment by Synexcu** [comment]
So, suppose I mapped the combat node coords with DMM Customization enabled in KC3, would it have the same result if I were to do it in an non-customized DMM environment (The same environment you used in your screenshot)? I would like to assume yes because we have set sort of like `anchor point` from the top left side (red circle) of the map, but I would like you to verify my assumption, just in case.

**Comment by XVs32** [comment]
Yep, you can have your KC3 Customization turned on when mapping the coords.
All kcauto needs is the offset from top left of the game window to the node.
The absolute coords of the game window is detected on kcauto's start up (Around the `Finding Browser` and `Finding kancolle` log).
